Framework/Beamline/src/ComponentInfo.cpp

```diff
@@ -113,21 +114,46 @@ ComponentInfo::componentsInSubtree(const size_t componentIndex) const {
 size_t ComponentInfo::size() const { return m_size; }
 
 Eigen::Vector3d ComponentInfo::position(const size_t componentIndex) const {
+  checkNoTimeDependence();
```

Comment:
Agreed. This would only be relevant if there was scanning of a non-detector component. Based on our previous discussion about scanning samples/sources (which we agreed to forbid) I don't see that happening. However I'd like to keep this check in place until I am confident that we are reasonably able to verify that we cannot have a scanning ComponentInfo without a scanning DetectorInfo.